### ISSUE TITLE
Lint/UselessAccessModifier should take scope into account

### ISSUE THREAD

**Issue by smaximov** [bbatsov/rubocop#3764]
Consider the following example:

``` ruby
class Foo # rubocop:disable Style/Documentation
  # ...

  contract do
    # ...

    private

    def private_contract_method(args)
      # process args
    end
  end

  private

  def private_foo_method(args)
    # process args
  end
end
```

Class method `::contract` here is used to generate some class with private methods, and it seems RuboCop cannot differentiate between the `private` access modifier inside the block passed to `::contract` and the one used inside the `Foo` class.

## Steps to reproduce the problem

Run RuboCop with the abovementioned snippet:

```
$ rubocop --display-cop-names test.rb
```

## Expected behavior

RuboCop finishes without errors or warnings.

## Actual behavior

RuboCop outputs the "Lint/UselessAccessModifier" warning:

```
Inspecting 1 file
W

Offenses:

test.rb:14:3: W: Lint/UselessAccessModifier: Useless private access modifier.
  private
  ^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.46.0 (using Parser 2.3.2.0, running on ruby 2.3.1 x86_64-linux)
```

**Comment by smaximov** [comment]
Just searched through closed issues and found #3422, sorry.
